src/main/java/gov/cabinetoffice/gap/adminbackend/services/SchemeEditorService.java

```diff
@@ -81,5 +87,30 @@ private List<String> getEmailsFromUserSubBatch(final List<String> userSubs, fina
                 awsEncryptionService.decryptField(item.emailAddress())).toList();
     }
 
+    public SchemeEntity addEditorToScheme(Integer schemeId, String editorEmailAddress, String jwt) {
+        AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
+        SchemeEntity scheme = this.schemeRepo.findById(schemeId).orElseThrow(EntityNotFoundException::new);
+        List<GrantAdmin> existingEditors = scheme.getGrantAdmins();
+        GrantAdmin editorToAdd;
+        try {
+            editorToAdd = userService.getGrantAdminIdFromUserServiceEmail(editorEmailAddress, jwt);
+        } catch (Exception e) {
+            throw new FieldViolationException("editorEmailAddress", "This account does not have an 'Administrator' account.");
+        }
+
+        if (existingEditors.stream().anyMatch(editor -> editor.getId().equals(editorToAdd.getId()))) {
+            throw new FieldViolationException("editorEmailAddress", "This email address is already an editor for this scheme");
+        }
+
+        if (!scheme.getCreatedBy().equals(session.getGrantAdminId())) {
```

Comment:
Worth speaking to Dylan about this check as he's obviously working on some nice AOP driven code to reduce the need for these.

Reply:
Yeah I concur, https://technologyprogramme.atlassian.net/browse/GAP-2448 is removing the need for this check at the service level. Plus we now need to check scheme editors rather than scheme owners

Reply:
Can remove this check